extensions/amp-jwplayer/0.1/amp-jwplayer.js

```
@@ -283,28 +304,42 @@ class AmpJWPlayer extends AMP.BaseElement {
 
   /** @override */
   preimplementsMediaSessionAPI() {
-    return true;
+    return false;
+  }
+
+  isSafariOrIos() {
+    const platform = Services.platformFor(this.win);
+
+    return platform.isSafari() || platform.isIos()
   }
 
   /** @override */
   fullscreenEnter() {
-    if (!this.iframe_ || this.fullscreen_) {
+    if (!this.iframe_) {
       return;
     }
-    this.sendCommand_('setFullscreen', true);
+    if (this.isSafariOrIos()) {
+      this.sendCommand_('setFullscreen', true);
+    } else {
+      fullscreenEnter(dev().assertElement(this.iframe_));
```

dev command here? seems like this wouldn't work in a prod build

---

`dev().assertElement(this.iframe_)` becomes `this.iframe_ `in prod. It's used in Google's official amp-video component, as well as Dailymotions and Brightcove's. It's about adding error signaling in dev for situations that would fail silently (ie. unlayout during fullscreen).